### PR TITLE
Implement Controls + Make BaseOverlayView abstract

### PR DESCRIPTION
**Pull request by Nour-Cheour10**

### CODE REVIEW

File: ipyopenlayers/example.py [hunk 1]
```diff
@@ -24,28 +24,62 @@ class TileLayer(Widget):
...
+   
    popup_content = Unicode('').tag(sync=True)
 
+class BaseControl(DOMWidget):
+   _model_name = Unicode('BaseControlModel').tag(sync=True)
+   _model_module = Unicode(module_name).tag(sync=True)
+   _model_module_version = Unicode(module_version).tag(sync=True)
+   _view_name = Unicode('BaseControlView').tag(sync=True)
+   _view_module = Unicode(module_name).tag(sync=True)
+   _view_module_version = Unicode(module_version).tag(sync=True)
```
Comment (martinRenou) [inline ipyopenlayers/example.py]: We probably don't need `_model_name` and `_view_name` here

```suggestion
   _model_module = Unicode(module_name).tag(sync=True)
   _model_module_version = Unicode(module_version).tag(sync=True)
   _view_module = Unicode(module_name).tag(sync=True)
   _view_module_version = Unicode(module_version).tag(sync=True)
```

File: src/basecontrol.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  render() {
+    super.render();
+    this.createObj();
+  }
+  createObj() {
+    throw new Error('Not implemented');
+  }
+}
+*/
```
Comment (martinRenou) [inline src/basecontrol.ts]:
```suggestion
```

File: src/imageoverlay.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  static model_module_version = MODULE_VERSION;
+  static view_name = 'ImageOverlayView';
+  static view_module = MODULE_NAME;
+  static view_module_version = MODULE_VERSION;
+}
+
+export class ImageOverlayView extends BaseOverlayView {
+  render() {
+    super.render();
+    console.log('hello');
```
Comment (martinRenou) [inline src/imageoverlay.ts]:
```suggestion
```

File: src/fullscreen.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  static view_name = 'FullScreenView';
+  static view_module = MODULE_NAME;
+  static view_module_version = MODULE_VERSION;
+}
+export class FullScreenView extends BaseControlView {
+  render() {
+    super.render();
+    this.element = document.createElement('div');
+    this.createObj();
+  }
```
Comment (martinRenou) [inline src/fullscreen.ts]: `this.element` doesn't seem to be used, so we can probably remove the render method entirely and just inherit from the one in the base?

```suggestion
```

File: src/mouseposition.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  static view_name = 'MousePositionView';
+  static view_module = MODULE_NAME;
+  static view_module_version = MODULE_VERSION;
+}
+export class MousePositionView extends BaseControlView {
+  render() {
+    super.render();
+    this.element = document.createElement('div');
+    this.createObj();
+  }
```
Comment (martinRenou) [inline src/mouseposition.ts]: Same comment as above

```suggestion
```

File: src/scaleline.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  static view_name = 'ScaleLineView';
+  static view_module = MODULE_NAME;
+  static view_module_version = MODULE_VERSION;
+}
+export class ScaleLineView extends BaseControlView {
+  render() {
+    super.render();
+    this.element = document.createElement('div');
+    this.createObj();
+  }
```
Comment (martinRenou) [inline src/scaleline.ts]: Same

File: src/tilelayer.ts [hunk 7]
```diff
@@ -32,6 +35,7 @@ export class TileLayerModel extends WidgetModel {
 }
 
 export class TileLayerView extends WidgetView {
+  map: MapView;
```
Comment (martinRenou) [inline src/tilelayer.ts]: Do we really need this? It seems it's never used

File: src/zoomslider.ts [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  static view_name = 'ZoomSliderView';
+  static view_module = MODULE_NAME;
+  static view_module_version = MODULE_VERSION;
+}
+export class ZoomSliderView extends BaseControlView {
+  render() {
+    super.render();
+    this.element = document.createElement('div');
+    this.createObj();
+  }
```
Comment (martinRenou) [inline src/zoomslider.ts]: Same

File: ipyopenlayers/example.py [hunk 9]
```diff
@@ -24,28 +24,60 @@ class TileLayer(Widget):
...
+   
    popup_content = Unicode('').tag(sync=True)
 
+class BaseControl(DOMWidget):
+   _model_module = Unicode(module_name).tag(sync=True)
+   _model_module_version = Unicode(module_version).tag(sync=True)
+   _view_module = Unicode(module_name).tag(sync=True)
+   _view_module_version = Unicode(module_version).tag(sync=True)
+   
+   control_name = Unicode().tag(sync=True)
```
Comment (martinRenou) [inline ipyopenlayers/example.py]: It seems this is not used?

**Review by martinRenou — CHANGES_REQUESTED** [review]
Thanks! I just have one comment, otherwiser I think it's good!

File: ipyopenlayers/example.py (hunk 9 above)
Comment (Nour-Cheour10) [reply ipyopenlayers/example.py:59]: Yes, it's not use. 
Thanks, I just took it off.

Comment (Nour-Cheour10) [reply ipyopenlayers/example.py]: Yes, it's not.
Thanks, I just took it off.

**Comment by martinRenou** [comment]
Thanks!

**Pull request merged**